textdocument.vala

```diff
@@ -0,0 +1,40 @@
+class Vls.TextDocument : Object {
+    public weak Compilation compilation { get; private set; }
+    private string filename;
+
+    public Vala.SourceFile file { get; private set; }
+    public string uri { get; private set; }
```

Comment:
why do we need the URI if we have the filename?

Reply:
It's just extra information used by the various debug statements for better readability. It could also be useful if, in the future, we support files that aren't on disk.